pytest_checkdocs/__init__.py

```diff
@@ -57,7 +59,12 @@ def system_message(reporter, level, message, *children, **kwargs):
         docutils.utils.Reporter.system_message = orig
 
     def get_long_description(self):
-        return Description.from_md(ensure_clean(pep517.meta.load('.').metadata))
+        builder = build.ProjectBuilder('.')
```

Comment:
This code is doing too much. It shouldn't be up to the caller to write all the logic with context managers and temporary directories. I'd really like for some project (pep517 or build or whatever) to provide a routine (like `pep517.meta.load`) that in one line takes a project path and returns the metadata in memory (no context managers, no temporary files to clean up, just a handle to the metadata). I want this project to serve as a canonical example of loading metadata for a project on disk and I don't want them to have to copy all these lines (and re-write them to integrate with their use-case).

Reply:
I haven't looked into the implementation, but I'm a little worried that I can't tell what's happening here. I see a "ProjectBuilder" and a "metadata_path", but I don't see any invocations to build the project or its metadata, which indicates that it must be doing it implicitly. How do I know that it's not also doing something else implicitly (building the project, for example). I definitely don't want to be building the whole project just to get the metadata (performance is bad enough that pip/pep517/build has to build the project in an isolated environment to generate the metadata).

Reply:
That'd require us to bring in `pathlib2` or reimplement a `pathlib.Path`-like object. We have talked about adding a plumbing module, this routine would be a good candidate to add, but for now we don't have anything like that and do not want to add this to the core API.

Reply:
> I haven't looked into the implementation, but I'm a little worried that I can't tell what's happening here. I see a "ProjectBuilder" and a "metadata_path", but I don't see any invocations to build the project or its metadata, which indicates that it must be doing it implicitly. How do I know that it's not also doing something else implicitly (building the project, for example). I definitely don't want to be building the whole project just to get the metadata (performance is bad enough that pip/pep517/build has to build the project in an isolated environment to generate the metadata).

https://pypa-build.readthedocs.io/en/stable/api.html#build.ProjectBuilder.metadata_path

It takes the shortest path, as recommended by PEP 517. This calls `prepare_metadata_for_build_wheel` and falls back to `build_wheel` when it is not present.